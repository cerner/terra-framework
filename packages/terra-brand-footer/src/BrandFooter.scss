:local {
  .brand-footer {
    margin: none;
  }

  .nav {
    background-color: var(--terra-brand-footer-nav-background-color, #f4f4f4);
    color: var(--terra-brand-footer-nav-color, #64696c);
    display: block;
    padding: 0.7142857142857143rem 1.4285714285714286rem;
  }

  .menu {
    list-style: none;
    margin: 0;
    padding-left: 0;
    padding-right: 0;
  }

  .list-item {
    display: inline;
  }


  .list-item:first-child::before {
    content: '';
    padding: 0;
  }

  .list-item::before {
    content: '/';
    padding: 0.7142857142857143rem;
  }


  .link {
    color: var(--terra-brand-footer-link-color, #0059a8);
    text-decoration: none;
  }

  .link:hover {
    color: var(--terra-brand-footer-link-hover-color, #001f67);
  }

  .link:active {
    color: var(--terra-brand-footer-link-active-color, #001f67);
  }

  .footer-content {
    background-color: var(--terra-brand-footer-footer-content-background-color, #0079be);
    color: var(--terra-brand-footer-footer-content-color, #fff);
    padding: 1.4285714285714286rem;
  }

  .content-top {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
  }

  .content-bottom {
    margin-top: 0.7142857142857143rem;
  }
}
