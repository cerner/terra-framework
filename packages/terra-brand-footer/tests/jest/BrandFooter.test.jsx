import React from 'react';
import BrandFooter from '../../src/BrandFooter';

describe('BrandFooter', () => {
  const defaultRender = <BrandFooter />;

  // Snapshot Tests
  it('should render a default component', () => {
    const wrapper = shallow(<BrandFooter />);
    expect(wrapper).toMatchSnapshot();
  });

  // Prop Tests
  it('should render with provided links', () => {
    const wrapper = shallow(<BrandFooter
      sections={[
        {
          links: [
            { text: 'Cerner Home', href: 'https://www.cerner.com/' },
            { text: 'Cerner Code', href: 'https://code.cerner.com/', target: '_blank' },
          ],
        },
      ]}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with the links prop', () => {
    const wrapper = shallow(<BrandFooter
      links={[
        { text: 'Cerner Home', href: 'https://www.cerner.com/' },
        { text: 'Cerner Code', href: 'https://code.cerner.com/', target: '_blank' },
      ]}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the sections prop instead of links prop', () => {
    const wrapper = shallow(<BrandFooter
      sections={[
        {
          links: [
            { text: 'Sections Prop link', href: 'https://www.cerner.com/' },
          ],
        },
      ]}
      links={[
        { text: 'Links Prop Link', href: 'https://code.cerner.com/' },
      ]}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a section header', () => {
    const wrapper = shallow(<BrandFooter
      sections={[
        {
          headerText: 'Cerner Links',
        },
      ]}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with a section header and provided links', () => {
    const wrapper = shallow(<BrandFooter
      sections={[
        {
          headerText: 'Cerner Links',
          links: [
            { text: 'Cerner Home', href: 'https://www.cerner.com/' },
            { text: 'Cerner Code', href: 'https://code.cerner.com/', target: '_blank' },
          ],
        },
      ]}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with multiple link groups', () => {
    const wrapper = shallow(<BrandFooter
      sections={[
        {
          headerText: 'Cerner Links',
          links: [
            { text: 'Cerner Home', href: 'https://www.cerner.com/' },
            { text: 'Cerner Code', href: 'https://code.cerner.com/', target: '_blank' },
          ],
        },
        {
          headerText: 'Cerner Links Again',
          links: [
            { text: 'Cerner Home Also', href: 'https://www.cerner.com/' },
            { text: 'Cerner Code As Well', href: 'https://code.cerner.com/', target: '_blank' },
          ],
        },
      ]}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with provided content left', () => {
    const wrapper = shallow(<BrandFooter
      contentLeft={
        <img src="https://github.com/cerner/terra-core/raw/master/terra.png" alt="terra-ui" height="40px" width="40px" />
        }
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with provided content right', () => {
    const wrapper = shallow(<BrandFooter
      contentRight={(
        <div>
          <p><small>Powered by</small></p>
          <p><b>React.js</b></p>
        </div>
)}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with provided content bottom', () => {
    const wrapper = shallow(<BrandFooter
      contentBottom={(
        <p>
          <small>Copyright &copy; 2018 Cerner Corporation.  All Rights Reserved.</small>
        </p>
)}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with provided links and content left', () => {
    const component = (
      <BrandFooter
        sections={[
          {
            links: [
              { text: 'Cerner Home', href: 'https://www.cerner.com/' },
              { text: 'Cerner Code', href: 'https://code.cerner.com/', target: '_blank' },
            ],
          },
        ]}
        contentLeft={
          <img src="https://github.com/cerner/terra-core/raw/master/terra.png" alt="terra-ui" height="40px" width="40px" />
        }
      />
    );
    const wrapper = shallow(component);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with provided links and content right', () => {
    const component = (
      <BrandFooter
        sections={[
          {
            links: [
              { text: 'Cerner Home', href: 'https://www.cerner.com/' },
              { text: 'Cerner Code', href: 'https://code.cerner.com/', target: '_blank' },
            ],
          },
        ]}
        contentRight={(
          <div>
            <p><small>Powered by</small></p>
            <p><b>React.js</b></p>
          </div>
)}
      />
    );
    const wrapper = shallow(component);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with provided links and content bottom', () => {
    const component = (
      <BrandFooter
        sections={[
          {
            links: [
              { text: 'Cerner Home', href: 'https://www.cerner.com/' },
              { text: 'Cerner Code', href: 'https://code.cerner.com/', target: '_blank' },
            ],
          },
        ]}
        contentBottom={(
          <p>
            <small>Copyright &copy; 2018 Cerner Corporation.  All Rights Reserved.</small>
          </p>
)}
      />
    );
    const wrapper = shallow(component);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with provided links, content left, content bottom and content right', () => {
    const component = (
      <BrandFooter
        sections={[
          {
            links: [
              { text: 'Cerner Home', href: 'https://www.cerner.com/' },
              { text: 'Cerner Code', href: 'https://code.cerner.com/', target: '_blank' },
            ],
          },
        ]}
        contentLeft={
          <img src="https://github.com/cerner/terra-core/raw/master/terra.png" alt="terra-ui" height="40px" width="40px" />
      }
        contentBottom={(
          <p>
            <small>Copyright &copy; 2018 Cerner Corporation.  All Rights Reserved.</small>
          </p>
)}
        contentRight={(
          <div>
            <p><small>Powered by</small></p>
            <p><b>React.js</b></p>
          </div>
)}
      />
    );
    const wrapper = shallow(component);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with provided links and content left', () => {
    const component = (
      <BrandFooter
        links={[
          { text: 'Cerner Home', href: 'https://www.cerner.com/' },
          { text: 'Cerner Code', href: 'https://code.cerner.com/', target: '_blank' },
        ]}
        contentLeft={
          <img src="https://github.com/cerner/terra-core/raw/master/terra.png" alt="terra-ui" height="40px" width="40px" />
      }
      />
    );
    const wrapper = shallow(component);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with provided links and content right', () => {
    const wrapper = shallow(<BrandFooter
      links={[
        { text: 'Cerner Home', href: 'https://www.cerner.com/' },
        { text: 'Cerner Code', href: 'https://code.cerner.com/', target: '_blank' },
      ]}
      contentRight={(
        <div>
          <p><small>Powered by</small></p>
          <p><b>React.js</b></p>
        </div>
)}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with provided links and content bottom', () => {
    const wrapper = shallow(<BrandFooter
      links={[
        { text: 'Cerner Home', href: 'https://www.cerner.com/' },
        { text: 'Cerner Code', href: 'https://code.cerner.com/', target: '_blank' },
      ]}
      contentBottom={(
        <p>
          <small>Copyright &copy; 2018 Cerner Corporation.  All Rights Reserved.</small>
        </p>
)}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with provided links, content left, content bottom and content right', () => {
    const wrapper = shallow(<BrandFooter
      links={[
        { text: 'Cerner Home', href: 'https://www.cerner.com/' },
        { text: 'Cerner Code', href: 'https://code.cerner.com/', target: '_blank' },
      ]}
      contentLeft={
        <img src="https://github.com/cerner/terra-core/raw/master/terra.png" alt="terra-ui" height="40px" width="40px" />
        }
      contentBottom={(
        <p>
          <small>Copyright &copy; 2018 Cerner Corporation.  All Rights Reserved.</small>
        </p>
)}
      contentRight={(
        <div>
          <p><small>Powered by</small></p>
          <p><b>React.js</b></p>
        </div>
)}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with content bottom and content left', () => {
    const wrapper = shallow(<BrandFooter
      contentBottom={(
        <p>
          <small>Copyright &copy; 2018 Cerner Corporation.  All Rights Reserved.</small>
        </p>
)}
      contentLeft={
        <img src="https://github.com/cerner/terra-core/raw/master/terra.png" alt="terra-ui" height="40px" width="40px" />
        }
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with content bottom and content right', () => {
    const wrapper = shallow(<BrandFooter
      contentBottom={(
        <p>
          <small>Copyright &copy; 2018 Cerner Corporation.  All Rights Reserved.</small>
        </p>
)}
      contentRight={(
        <div>
          <p><small>Powered by</small></p>
          <p><b>React.js</b></p>
        </div>
)}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with content bottom, content right and content left', () => {
    const wrapper = shallow(<BrandFooter
      contentLeft={
        <img src="https://github.com/cerner/terra-core/raw/master/terra.png" alt="terra-ui" height="40px" width="40px" />
        }
      contentBottom={(
        <p>
          <small>Copyright &copy; 2018 Cerner Corporation.  All Rights Reserved.</small>
        </p>
)}
      contentRight={(
        <div>
          <p><small>Powered by</small></p>
          <p><b>React.js</b></p>
        </div>
)}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with content right and content left', () => {
    const wrapper = shallow(<BrandFooter
      contentLeft={
        <img src="https://github.com/cerner/terra-core/raw/master/terra.png" alt="terra-ui" height="40px" width="40px" />
        }
      contentRight={(
        <div>
          <p><small>Powered by</small></p>
          <p><b>React.js</b></p>
        </div>
)}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  // Structure Tests
  it('should have the class brand-footer', () => {
    const wrapper = shallow(defaultRender);
    expect(wrapper.prop('className')).toContain('brand-footer');
  });

  it('correctly applies the theme context className', () => {
    jest.spyOn(React, 'useContext')
      .mockReturnValue({
        className: 'orion-fusion-theme',
      });
    const wrapper = shallow(<BrandFooter />);
    expect(wrapper).toMatchSnapshot();
  });
});
