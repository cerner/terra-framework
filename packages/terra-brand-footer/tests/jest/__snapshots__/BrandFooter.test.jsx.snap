// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrandFooter correctly applies the theme context className 1`] = `
<footer
  className="brand-footer orion-fusion-theme"
  role="contentinfo"
>
  <div
    className="footer-content"
  >
    <div
      className="content-top"
    />
    <div
      className="content-bottom"
    />
  </div>
</footer>
`;

exports[`BrandFooter should render a default component 1`] = `
<footer
  className="brand-footer"
  role="contentinfo"
>
  <div
    className="footer-content"
  >
    <div
      className="content-top"
    />
    <div
      className="content-bottom"
    />
  </div>
</footer>
`;

exports[`BrandFooter should render a section header 1`] = `
<footer
  className="brand-footer"
  role="contentinfo"
>
  <nav
    className="nav nav-horizontal"
  >
    <section
      className="navigation-section"
      key="0"
    >
      <h3
        className="list-header"
        key="Cerner Links"
      >
        Cerner Links
      </h3>
      <ul
        className="menu"
      />
    </section>
  </nav>
  <div
    className="footer-content"
  >
    <div
      className="content-top"
    />
    <div
      className="content-bottom"
    />
  </div>
</footer>
`;

exports[`BrandFooter should render the sections prop instead of links prop 1`] = `
<footer
  className="brand-footer"
  role="contentinfo"
>
  <nav
    className="nav nav-horizontal"
  >
    <section
      className="navigation-section"
      key="0"
    >
      <ul
        className="menu"
      >
        <li
          className="list-item"
          key="Sections Prop linkhttps://www.cerner.com/"
        >
          <a
            className="link"
            href="https://www.cerner.com/"
          >
            Sections Prop link
          </a>
        </li>
      </ul>
    </section>
  </nav>
  <div
    className="footer-content"
  >
    <div
      className="content-top"
    />
    <div
      className="content-bottom"
    />
  </div>
</footer>
`;

exports[`BrandFooter should render with a section header and provided links 1`] = `
<footer
  className="brand-footer"
  role="contentinfo"
>
  <nav
    className="nav nav-horizontal"
  >
    <section
      className="navigation-section"
      key="0"
    >
      <h3
        className="list-header"
        key="Cerner Links"
      >
        Cerner Links
      </h3>
      <ul
        className="menu"
      >
        <li
          className="list-item"
          key="Cerner Homehttps://www.cerner.com/"
        >
          <a
            className="link"
            href="https://www.cerner.com/"
          >
            Cerner Home
          </a>
        </li>
        <li
          className="list-item"
          key="Cerner Codehttps://code.cerner.com/"
        >
          <span
            aria-hidden={true}
            className="separator"
          >
            /
          </span>
          <a
            className="link"
            href="https://code.cerner.com/"
            target="_blank"
          >
            Cerner Code
          </a>
        </li>
      </ul>
    </section>
  </nav>
  <div
    className="footer-content"
  >
    <div
      className="content-top"
    />
    <div
      className="content-bottom"
    />
  </div>
</footer>
`;

exports[`BrandFooter should render with content bottom and content left 1`] = `
<footer
  className="brand-footer"
  role="contentinfo"
>
  <div
    className="footer-content"
  >
    <div
      className="content-top"
    >
      <img
        alt="terra-ui"
        height="40px"
        src="https://github.com/cerner/terra-core/raw/master/terra.png"
        width="40px"
      />
    </div>
    <div
      className="content-bottom"
    >
      <p>
        <small>
          Copyright © 2018 Cerner Corporation.  All Rights Reserved.
        </small>
      </p>
    </div>
  </div>
</footer>
`;

exports[`BrandFooter should render with content bottom and content right 1`] = `
<footer
  className="brand-footer"
  role="contentinfo"
>
  <div
    className="footer-content"
  >
    <div
      className="content-top"
    >
      <div>
        <p>
          <small>
            Powered by
          </small>
        </p>
        <p>
          <b>
            React.js
          </b>
        </p>
      </div>
    </div>
    <div
      className="content-bottom"
    >
      <p>
        <small>
          Copyright © 2018 Cerner Corporation.  All Rights Reserved.
        </small>
      </p>
    </div>
  </div>
</footer>
`;

exports[`BrandFooter should render with content bottom, content right and content left 1`] = `
<footer
  className="brand-footer"
  role="contentinfo"
>
  <div
    className="footer-content"
  >
    <div
      className="content-top"
    >
      <img
        alt="terra-ui"
        height="40px"
        src="https://github.com/cerner/terra-core/raw/master/terra.png"
        width="40px"
      />
      <div>
        <p>
          <small>
            Powered by
          </small>
        </p>
        <p>
          <b>
            React.js
          </b>
        </p>
      </div>
    </div>
    <div
      className="content-bottom"
    >
      <p>
        <small>
          Copyright © 2018 Cerner Corporation.  All Rights Reserved.
        </small>
      </p>
    </div>
  </div>
</footer>
`;

exports[`BrandFooter should render with content right and content left 1`] = `
<footer
  className="brand-footer"
  role="contentinfo"
>
  <div
    className="footer-content"
  >
    <div
      className="content-top"
    >
      <img
        alt="terra-ui"
        height="40px"
        src="https://github.com/cerner/terra-core/raw/master/terra.png"
        width="40px"
      />
      <div>
        <p>
          <small>
            Powered by
          </small>
        </p>
        <p>
          <b>
            React.js
          </b>
        </p>
      </div>
    </div>
    <div
      className="content-bottom"
    />
  </div>
</footer>
`;

exports[`BrandFooter should render with multiple link groups 1`] = `
<footer
  className="brand-footer"
  role="contentinfo"
>
  <nav
    className="nav nav-horizontal"
  >
    <section
      className="navigation-section"
      key="0"
    >
      <h3
        className="list-header"
        key="Cerner Links"
      >
        Cerner Links
      </h3>
      <ul
        className="menu"
      >
        <li
          className="list-item"
          key="Cerner Homehttps://www.cerner.com/"
        >
          <a
            className="link"
            href="https://www.cerner.com/"
          >
            Cerner Home
          </a>
        </li>
        <li
          className="list-item"
          key="Cerner Codehttps://code.cerner.com/"
        >
          <span
            aria-hidden={true}
            className="separator"
          >
            /
          </span>
          <a
            className="link"
            href="https://code.cerner.com/"
            target="_blank"
          >
            Cerner Code
          </a>
        </li>
      </ul>
    </section>
    <section
      className="navigation-section"
      key="1"
    >
      <h3
        className="list-header"
        key="Cerner Links Again"
      >
        Cerner Links Again
      </h3>
      <ul
        className="menu"
      >
        <li
          className="list-item"
          key="Cerner Home Alsohttps://www.cerner.com/"
        >
          <a
            className="link"
            href="https://www.cerner.com/"
          >
            Cerner Home Also
          </a>
        </li>
        <li
          className="list-item"
          key="Cerner Code As Wellhttps://code.cerner.com/"
        >
          <span
            aria-hidden={true}
            className="separator"
          >
            /
          </span>
          <a
            className="link"
            href="https://code.cerner.com/"
            target="_blank"
          >
            Cerner Code As Well
          </a>
        </li>
      </ul>
    </section>
  </nav>
  <div
    className="footer-content"
  >
    <div
      className="content-top"
    />
    <div
      className="content-bottom"
    />
  </div>
</footer>
`;

exports[`BrandFooter should render with provided content bottom 1`] = `
<footer
  className="brand-footer"
  role="contentinfo"
>
  <div
    className="footer-content"
  >
    <div
      className="content-top"
    />
    <div
      className="content-bottom"
    >
      <p>
        <small>
          Copyright © 2018 Cerner Corporation.  All Rights Reserved.
        </small>
      </p>
    </div>
  </div>
</footer>
`;

exports[`BrandFooter should render with provided content left 1`] = `
<footer
  className="brand-footer"
  role="contentinfo"
>
  <div
    className="footer-content"
  >
    <div
      className="content-top"
    >
      <img
        alt="terra-ui"
        height="40px"
        src="https://github.com/cerner/terra-core/raw/master/terra.png"
        width="40px"
      />
    </div>
    <div
      className="content-bottom"
    />
  </div>
</footer>
`;

exports[`BrandFooter should render with provided content right 1`] = `
<footer
  className="brand-footer"
  role="contentinfo"
>
  <div
    className="footer-content"
  >
    <div
      className="content-top"
    >
      <div>
        <p>
          <small>
            Powered by
          </small>
        </p>
        <p>
          <b>
            React.js
          </b>
        </p>
      </div>
    </div>
    <div
      className="content-bottom"
    />
  </div>
</footer>
`;

exports[`BrandFooter should render with provided links 1`] = `
<footer
  className="brand-footer"
  role="contentinfo"
>
  <nav
    className="nav nav-horizontal"
  >
    <section
      className="navigation-section"
      key="0"
    >
      <ul
        className="menu"
      >
        <li
          className="list-item"
          key="Cerner Homehttps://www.cerner.com/"
        >
          <a
            className="link"
            href="https://www.cerner.com/"
          >
            Cerner Home
          </a>
        </li>
        <li
          className="list-item"
          key="Cerner Codehttps://code.cerner.com/"
        >
          <span
            aria-hidden={true}
            className="separator"
          >
            /
          </span>
          <a
            className="link"
            href="https://code.cerner.com/"
            target="_blank"
          >
            Cerner Code
          </a>
        </li>
      </ul>
    </section>
  </nav>
  <div
    className="footer-content"
  >
    <div
      className="content-top"
    />
    <div
      className="content-bottom"
    />
  </div>
</footer>
`;

exports[`BrandFooter should render with provided links and content bottom 1`] = `
<footer
  className="brand-footer"
  role="contentinfo"
>
  <nav
    className="nav nav-horizontal"
  >
    <section
      className="navigation-section"
      key="0"
    >
      <ul
        className="menu"
      >
        <li
          className="list-item"
          key="Cerner Homehttps://www.cerner.com/"
        >
          <a
            className="link"
            href="https://www.cerner.com/"
          >
            Cerner Home
          </a>
        </li>
        <li
          className="list-item"
          key="Cerner Codehttps://code.cerner.com/"
        >
          <span
            aria-hidden={true}
            className="separator"
          >
            /
          </span>
          <a
            className="link"
            href="https://code.cerner.com/"
            target="_blank"
          >
            Cerner Code
          </a>
        </li>
      </ul>
    </section>
  </nav>
  <div
    className="footer-content"
  >
    <div
      className="content-top"
    />
    <div
      className="content-bottom"
    >
      <p>
        <small>
          Copyright © 2018 Cerner Corporation.  All Rights Reserved.
        </small>
      </p>
    </div>
  </div>
</footer>
`;

exports[`BrandFooter should render with provided links and content bottom 2`] = `
<footer
  className="brand-footer"
  role="contentinfo"
>
  <nav
    className="nav nav-horizontal"
  >
    <section
      className="navigation-section"
      key="0"
    >
      <ul
        className="menu"
      >
        <li
          className="list-item"
          key="Cerner Homehttps://www.cerner.com/"
        >
          <a
            className="link"
            href="https://www.cerner.com/"
          >
            Cerner Home
          </a>
        </li>
        <li
          className="list-item"
          key="Cerner Codehttps://code.cerner.com/"
        >
          <span
            aria-hidden={true}
            className="separator"
          >
            /
          </span>
          <a
            className="link"
            href="https://code.cerner.com/"
            target="_blank"
          >
            Cerner Code
          </a>
        </li>
      </ul>
    </section>
  </nav>
  <div
    className="footer-content"
  >
    <div
      className="content-top"
    />
    <div
      className="content-bottom"
    >
      <p>
        <small>
          Copyright © 2018 Cerner Corporation.  All Rights Reserved.
        </small>
      </p>
    </div>
  </div>
</footer>
`;

exports[`BrandFooter should render with provided links and content left 1`] = `
<footer
  className="brand-footer"
  role="contentinfo"
>
  <nav
    className="nav nav-horizontal"
  >
    <section
      className="navigation-section"
      key="0"
    >
      <ul
        className="menu"
      >
        <li
          className="list-item"
          key="Cerner Homehttps://www.cerner.com/"
        >
          <a
            className="link"
            href="https://www.cerner.com/"
          >
            Cerner Home
          </a>
        </li>
        <li
          className="list-item"
          key="Cerner Codehttps://code.cerner.com/"
        >
          <span
            aria-hidden={true}
            className="separator"
          >
            /
          </span>
          <a
            className="link"
            href="https://code.cerner.com/"
            target="_blank"
          >
            Cerner Code
          </a>
        </li>
      </ul>
    </section>
  </nav>
  <div
    className="footer-content"
  >
    <div
      className="content-top"
    >
      <img
        alt="terra-ui"
        height="40px"
        src="https://github.com/cerner/terra-core/raw/master/terra.png"
        width="40px"
      />
    </div>
    <div
      className="content-bottom"
    />
  </div>
</footer>
`;

exports[`BrandFooter should render with provided links and content left 2`] = `
<footer
  className="brand-footer"
  role="contentinfo"
>
  <nav
    className="nav nav-horizontal"
  >
    <section
      className="navigation-section"
      key="0"
    >
      <ul
        className="menu"
      >
        <li
          className="list-item"
          key="Cerner Homehttps://www.cerner.com/"
        >
          <a
            className="link"
            href="https://www.cerner.com/"
          >
            Cerner Home
          </a>
        </li>
        <li
          className="list-item"
          key="Cerner Codehttps://code.cerner.com/"
        >
          <span
            aria-hidden={true}
            className="separator"
          >
            /
          </span>
          <a
            className="link"
            href="https://code.cerner.com/"
            target="_blank"
          >
            Cerner Code
          </a>
        </li>
      </ul>
    </section>
  </nav>
  <div
    className="footer-content"
  >
    <div
      className="content-top"
    >
      <img
        alt="terra-ui"
        height="40px"
        src="https://github.com/cerner/terra-core/raw/master/terra.png"
        width="40px"
      />
    </div>
    <div
      className="content-bottom"
    />
  </div>
</footer>
`;

exports[`BrandFooter should render with provided links and content right 1`] = `
<footer
  className="brand-footer"
  role="contentinfo"
>
  <nav
    className="nav nav-horizontal"
  >
    <section
      className="navigation-section"
      key="0"
    >
      <ul
        className="menu"
      >
        <li
          className="list-item"
          key="Cerner Homehttps://www.cerner.com/"
        >
          <a
            className="link"
            href="https://www.cerner.com/"
          >
            Cerner Home
          </a>
        </li>
        <li
          className="list-item"
          key="Cerner Codehttps://code.cerner.com/"
        >
          <span
            aria-hidden={true}
            className="separator"
          >
            /
          </span>
          <a
            className="link"
            href="https://code.cerner.com/"
            target="_blank"
          >
            Cerner Code
          </a>
        </li>
      </ul>
    </section>
  </nav>
  <div
    className="footer-content"
  >
    <div
      className="content-top"
    >
      <div>
        <p>
          <small>
            Powered by
          </small>
        </p>
        <p>
          <b>
            React.js
          </b>
        </p>
      </div>
    </div>
    <div
      className="content-bottom"
    />
  </div>
</footer>
`;

exports[`BrandFooter should render with provided links and content right 2`] = `
<footer
  className="brand-footer"
  role="contentinfo"
>
  <nav
    className="nav nav-horizontal"
  >
    <section
      className="navigation-section"
      key="0"
    >
      <ul
        className="menu"
      >
        <li
          className="list-item"
          key="Cerner Homehttps://www.cerner.com/"
        >
          <a
            className="link"
            href="https://www.cerner.com/"
          >
            Cerner Home
          </a>
        </li>
        <li
          className="list-item"
          key="Cerner Codehttps://code.cerner.com/"
        >
          <span
            aria-hidden={true}
            className="separator"
          >
            /
          </span>
          <a
            className="link"
            href="https://code.cerner.com/"
            target="_blank"
          >
            Cerner Code
          </a>
        </li>
      </ul>
    </section>
  </nav>
  <div
    className="footer-content"
  >
    <div
      className="content-top"
    >
      <div>
        <p>
          <small>
            Powered by
          </small>
        </p>
        <p>
          <b>
            React.js
          </b>
        </p>
      </div>
    </div>
    <div
      className="content-bottom"
    />
  </div>
</footer>
`;

exports[`BrandFooter should render with provided links, content left, content bottom and content right 1`] = `
<footer
  className="brand-footer"
  role="contentinfo"
>
  <nav
    className="nav nav-horizontal"
  >
    <section
      className="navigation-section"
      key="0"
    >
      <ul
        className="menu"
      >
        <li
          className="list-item"
          key="Cerner Homehttps://www.cerner.com/"
        >
          <a
            className="link"
            href="https://www.cerner.com/"
          >
            Cerner Home
          </a>
        </li>
        <li
          className="list-item"
          key="Cerner Codehttps://code.cerner.com/"
        >
          <span
            aria-hidden={true}
            className="separator"
          >
            /
          </span>
          <a
            className="link"
            href="https://code.cerner.com/"
            target="_blank"
          >
            Cerner Code
          </a>
        </li>
      </ul>
    </section>
  </nav>
  <div
    className="footer-content"
  >
    <div
      className="content-top"
    >
      <img
        alt="terra-ui"
        height="40px"
        src="https://github.com/cerner/terra-core/raw/master/terra.png"
        width="40px"
      />
      <div>
        <p>
          <small>
            Powered by
          </small>
        </p>
        <p>
          <b>
            React.js
          </b>
        </p>
      </div>
    </div>
    <div
      className="content-bottom"
    >
      <p>
        <small>
          Copyright © 2018 Cerner Corporation.  All Rights Reserved.
        </small>
      </p>
    </div>
  </div>
</footer>
`;

exports[`BrandFooter should render with provided links, content left, content bottom and content right 2`] = `
<footer
  className="brand-footer"
  role="contentinfo"
>
  <nav
    className="nav nav-horizontal"
  >
    <section
      className="navigation-section"
      key="0"
    >
      <ul
        className="menu"
      >
        <li
          className="list-item"
          key="Cerner Homehttps://www.cerner.com/"
        >
          <a
            className="link"
            href="https://www.cerner.com/"
          >
            Cerner Home
          </a>
        </li>
        <li
          className="list-item"
          key="Cerner Codehttps://code.cerner.com/"
        >
          <span
            aria-hidden={true}
            className="separator"
          >
            /
          </span>
          <a
            className="link"
            href="https://code.cerner.com/"
            target="_blank"
          >
            Cerner Code
          </a>
        </li>
      </ul>
    </section>
  </nav>
  <div
    className="footer-content"
  >
    <div
      className="content-top"
    >
      <img
        alt="terra-ui"
        height="40px"
        src="https://github.com/cerner/terra-core/raw/master/terra.png"
        width="40px"
      />
      <div>
        <p>
          <small>
            Powered by
          </small>
        </p>
        <p>
          <b>
            React.js
          </b>
        </p>
      </div>
    </div>
    <div
      className="content-bottom"
    >
      <p>
        <small>
          Copyright © 2018 Cerner Corporation.  All Rights Reserved.
        </small>
      </p>
    </div>
  </div>
</footer>
`;

exports[`BrandFooter should render with the links prop 1`] = `
<footer
  className="brand-footer"
  role="contentinfo"
>
  <nav
    className="nav nav-horizontal"
  >
    <section
      className="navigation-section"
      key="0"
    >
      <ul
        className="menu"
      >
        <li
          className="list-item"
          key="Cerner Homehttps://www.cerner.com/"
        >
          <a
            className="link"
            href="https://www.cerner.com/"
          >
            Cerner Home
          </a>
        </li>
        <li
          className="list-item"
          key="Cerner Codehttps://code.cerner.com/"
        >
          <span
            aria-hidden={true}
            className="separator"
          >
            /
          </span>
          <a
            className="link"
            href="https://code.cerner.com/"
            target="_blank"
          >
            Cerner Code
          </a>
        </li>
      </ul>
    </section>
  </nav>
  <div
    className="footer-content"
  >
    <div
      className="content-top"
    />
    <div
      className="content-bottom"
    />
  </div>
</footer>
`;
